libcxx/test/std/utilities/memory/unique.ptr/noexcept_operator_star.compile.pass.cpp

```diff
@@ -1,25 +0,0 @@
-//===----------------------------------------------------------------------===//
-//
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-//
-//===----------------------------------------------------------------------===//
-
-// UNSUPPORTED: c++03
-
-// unique_ptr
-
-// add_lvalue_reference_t<T> operator*() const noexcept(noexcept(*declval<pointer>()));
-
-// Dereferencing pointer directly in noexcept fails for a void pointer.  This
-// is not SFINAE-ed away leading to a hard error. The issue was originally
-// triggered by
-// test/std/utilities/memory/unique.ptr/iterator_concept_conformance.compile.pass.cpp
```

Comment:
This was the cause for adding this.

Reply:
Got it. FWIW this test passes after applying this patch locally. Perhaps it used to fail with an older compiler? I seem to recall older Clang instantiating `noexcept` too eagerly at some point in the past.